src/styles/index.scss

```diff
@@ -1,3 +1,83 @@
+$day-size: 100px;
+$gap-size: 1px;
+$font-size: 30px;
+$day-bg: #eee;
+$hover-bg: #ffbfcb;
+$days: (
+  mon: 1,
+  tue: 2,
+  wed: 3,
+  thu: 4,
+  fri: 5,
+  sat: 6,
+  sun: 7,
+);
+
 body {
+  height: 100vh;
   margin: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.calendar {
+  display: grid;
+  grid-template-columns: repeat(7, $day-size);
+  grid-auto-rows: $day-size;
+  gap: $gap-size;
+
+  @for $length from 28 through 30 {
+    &--month-length-#{$length} .calendar__day:nth-child(n + #{$length + 1}) {
+      display: none;
+    }
+  }
+}
+
+.calendar__day {
+  position: relative;
+  border: 1px solid black;
+  background-color: $day-bg;
+  box-sizing: border-box;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-family: Arial, sans-serif;
+  font-size: $font-size;
+  transition:
+    background-color 0.5s,
+    transform 0.5s;
+
+  &:hover {
+    cursor: pointer;
+    background-color: $hover-bg;
+    transform: translateY(-20px);
+  }
+
+  &::before {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    font-size: $font-size;
+    color: black;
+  }
+
+  @for $i from 1 through 31 {
+    &:nth-child(#{$i})::before {
+      content: '#{$i}';
+    }
+  }
+}
+
+@each $day, $position in $days {
+  .calendar--start-day-#{$day} {
+    .calendar__day:first-child {
+      grid-column-start: $position;
+    }
+  }
+}
+
+.calendar--month-length-31 .calendar__day:nth-child(n + 32) {
```

Comment:
This code is redundant because you have already handled the hiding of days for different month lengths in the loop from lines 30 to 34. Consider removing this to keep your code DRY (Don't Repeat Yourself).